File: Algorithm/Leetcode/0000_0100/0021_0030/0025.py
# 1221. Split a String in Balanced Strings

"""https://leetcode.com/problems/split-a-string-in-balanced-strings/submissions/"""


class Solution:
    def balancedStringSplit(self, s: str) -> int:
        return list(accumulate(1 if c == "R" else -1 for c in s)).count(0)

File: Algorithm/Leetcode/0000_0100/0021_0030/0025.cpp
// 1221. Split a String in Balanced Strings


class Solution {
public:
    int balancedStringSplit(string s) {
        		int output = {};

		char start = {};
		int step = {};
		for (auto& letter:s)
		{
			if (0 == start)
			{
				start = letter;
				step = 1;
			}
			else if (start == letter)
			{
				step++;
			}
			else
			{
				--step;
				if (0 == step)
				{
					++output;
					start = 0;
				}
			}
		}
		return output;
    }
};